### PR TITLE
Update external dependencies

### PR DESCRIPTION
**Pull request by tvdijen**
Needs testing, but the changelogs give no reason to expect any issues

### CODE REVIEW

**Comment by khlr** [comment]
Hello Tim!

Thanks for helping to keep up with the dependencies!

I think there's no problem in replacing pako 2.0.4 with 2.1.0.
Looking at the [diff](https://github.com/simplesamlphp/SAML-tracer/pull/86/files#diff-79a1cd8273a2cea160b5e75b35f19f7bb31e00384e9ae258c057550e2001e5d3) there seems to be literally a single change in `pako_inflate.min.js`: the version number 😁
This is hardly surprising surprising (if I don't overlook something) since [pako's changelog](https://github.com/nodeca/pako/blob/master/CHANGELOG.md) is quite easy to grasp.

But something seems to be wrong with highlight.js:

*main:*

image

*PR:*

image

Could you please have a look at this issue?

I used https://stubidp.sustainsys.com for testing.

**Review by khlr — CHANGES_REQUESTED** [review]
Hello Tim!

Thanks for helping to keep up with the dependencies!

I think there's no problem in replacing pako 2.0.4 with 2.1.0.
Looking at the [diff](https://github.com/simplesamlphp/SAML-tracer/pull/86/files#diff-79a1cd8273a2cea160b5e75b35f19f7bb31e00384e9ae258c057550e2001e5d3) there seems to be literally a single change in `pako_inflate.min.js`: the version number 😁
This is hardly surprising surprising (if I don't overlook something) since [pako's changelog](https://github.com/nodeca/pako/blob/master/CHANGELOG.md) is quite easy to grasp.

But something seems to be wrong with highlight.js:

*main:*

image

*PR:*

image

Could you please have a look at this issue?

I used https://stubidp.sustainsys.com for testing.

**Comment by tvdijen** [comment]
Hmm, I just realized I've seen this same issue [elsewhere](https://github.com/simplesamlphp/simplesamlphp-assets-base/issues/21) ... Seems they have changed some defaults or something

**Comment by khlr** [comment]
What's a bit strange: I can't reproduce the unwanted behaviour out of SAML-tracer. Have a look at this fiddle: https://jsfiddle.net/fkwb3p8e/1/

There's a token string assigned to a div's `textContent`-property. It's done pretty much the same way as SAML-tracer does it.
But in the fiddle the whole token gets highlighted correctly...

Any ideas?

**Comment by tvdijen** [comment]
The only possible explanation I found is in the highlight.js docs:

> We strongly recommend <pre><code> wrapping for code blocks. It's quite semantic and "just works" out of the box with zero fiddling. It is possible to use other HTML elements (or combos), but you may need to pay special attention to preserving linebreaks.

I think maybe that's that we're seeing in your screenshot..

**Comment by tvdijen** [comment]
I guess the highlight package has changed over time too...
We have the same problem in SimpleSAMLphp where we use highlight.js to highlight metadata XML.. The entire color scheme has changed during the upgrade, but nothing in the changelog that could possibly explain this.

**Comment by khlr** [comment]
I took the liberty to update hljs to version 11.9.0. Whatever was changed with this version... The highlighting works now in any case again as desired...

@tvdijen: Would you please have a look, too?

**Comment by khlr** [comment]
Hmmm... I just took a look at [the diff](https://github.com/simplesamlphp/SAML-tracer/pull/86/commits/f2e21bf0b5622000bb9d1dcc7c536a7efa182a95) in our repo from version 11.8.0 to 11.9.0:
Search for the string `registerLanguage("`.
Did you just include the JavaScript-package? When I downloaded hljs today, I included *HTTP* and *XML*. Those two packages are required for the highlighting in SAML-tracer. Maybe this was just the cause for the incorrect highlighting, as hljs tried to highlight the token as JavaScript?

**Comment by tvdijen** [comment]
I just googled to get an updated version of the highlight.min.js file.. I was not aware of any additional packages

**Comment by tvdijen** [comment]
Ah, I think I understand where I went wrong :')  Good to know this, because I was going to create a workflow for auto-updating our dependencies, but this means I have to build a custom highlight.min.js package

**Comment by khlr** [comment]
Ah ok :-)

[Here](https://highlightjs.org/download) you can define the included packages. That's where I picked XML and HTTP. I then downloaded the ZIP file and took the highlight.min.js file.

**Comment by khlr** [comment]
> I was going to create a workflow for auto-updating our dependencies

Very interesting! How are you going to achieve this? Using GitHub actions or something?

**Review by khlr — APPROVED** [review]
Regarding the PR: From my perspective everything seems to look ok and you can merge the PR 😊

**Comment by tvdijen** [comment]
Let me rephrase that:

My goal was to migrate the build-script to actions:
https://github.com/simplesamlphp/SAML-tracer/compare/main...feature/autobuild

Now, this will work because the NPM-package for Highlight.js includes all 190-something languages. I could slightly improve this by building my own highlight.min.js file based on `highlight.js/core` + the two language packs, but since this is an extension and not a package transfered over the line, the added value would be minimal.

**Comment by khlr** [comment]
That sounds really promising, Tim!

However I would suggest that we should discuss this in another issue resp. alongside another PR's discussion, do you agree?

Do you agree then in merging _this PR_ here?
